yarn-project/pxe/src/pxe.ts

```diff
@@ -73,14 +74,19 @@ import { AddressDataProvider } from './storage/address_data_provider/address_dat
 import { CapsuleDataProvider } from './storage/capsule_data_provider/capsule_data_provider.js';
 import { ContractDataProvider } from './storage/contract_data_provider/contract_data_provider.js';
 import { NoteDataProvider } from './storage/note_data_provider/note_data_provider.js';
-import {
-  type PrivateEvent,
-  PrivateEventDataProvider,
-} from './storage/private_event_data_provider/private_event_data_provider.js';
+import { PrivateEventDataProvider } from './storage/private_event_data_provider/private_event_data_provider.js';
 import { SyncDataProvider } from './storage/sync_data_provider/sync_data_provider.js';
 import { TaggingDataProvider } from './storage/tagging_data_provider/tagging_data_provider.js';
 import { Synchronizer } from './synchronizer/index.js';
 
+export type PrivateEvent = {
+  msgContent: Fr[];
```

Comment:
```suggestion
  packedEvent: Fr[];
```

Terminology nit. 'messages' are the things people send to one another (through various channels), they have a ciphertext (and corresponding plaintex) and some metadata (tx hash etc). Some messages are event messages, which are identified by the 'msg id' field in the message header inside the message plaintext. An event message contains the packed content (an array of fields), which can be decoded to the actual typed event data, and the selector.

Should this struct not have the selector btw?

Reply:
ah yes, it's a leaky abstraction indeed, will fix. and re: selector, yes, probably a good idea. I'm introducing an EventFilter next and I guess there will end up being some symmetry between that type and this one in its "final" form